app/controllers/v0/user_action_events_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module V0
+  class UserActionEventsController < ApplicationController
+    def index
+      user_verification = current_user.user_verification
+
+      start_date = params[:start_date].present? ? Date.parse(params[:start_date]) : 1.month.ago.to_date
+      end_date = params[:end_date].present? ? Date.parse(params[:end_date]) : Time.zone.today
+      page = params[:page] || 1
+      per_page = params[:per_page] || 10
+
+      @user_actions = UserAction
+                      .where(subject_user_verification_id: user_verification.id)
+                      .where(created_at: start_date..end_date)
+                      .includes(:user_action_event)
+                      .order(created_at: :desc)
+                      .page(page)
+                      .per_page(per_page)
+      render json: @user_actions, include: [:user_action_event]
```

Comment:
Could you create serializers for `user_actions` and `user_action_events`? this will format it in the JSON:API format which frontend more or less expects

Reply:
I was able to create and use the UserActionSerialzer based on [this](https://github.com/QultureRocks/active_model_serializers/blob/master/docs/general/rendering.md#2-for-a-resource-collection:~:text=2.%20For%20a%20resource%20collection%3A)